src/Views/FirmwareView.vala

```diff
@@ -0,0 +1,157 @@
+/*
+* Copyright (c) 2020 elementary, Inc. (https://elementary.io)
+*
+* This program is free software; you can redistribute it and/or
+* modify it under the terms of the GNU General Public
+* License as published by the Free Software Foundation; either
+* version 3 of the License, or (at your option) any later version.
+*
+* This program is distributed in the hope that it will be useful,
+* but WITHOUT ANY WARRANTY; without even the implied warranty of
+* MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+* General Public License for more details.
+*
+* You should have received a copy of the GNU General Public
+* License along with this program; if not, write to the
+* Free Software Foundation, Inc., 51 Franklin Street, Fifth Floor,
+* Boston, MA 02110-1301 USA
+*
+* Authored by: Marius Meisenzahl <mariusmeisenzahl@gmail.com>
+*/
+
+public class About.FirmwareView : Granite.SettingsPage {
+    private Gtk.Stack stack;
+    private Gtk.Grid grid;
+    private Gtk.Grid progress_view;
+    private Gtk.Grid no_devices_view;
+    private Gtk.ListBox update_list;
+
+    public FirmwareView () {
+        Object (
+            icon_name: "application-x-firmware",
+            title: _("Firmware")
+        );
+    }
+
+    construct {
+        var progress_alert_view = new Granite.Widgets.AlertView (
+            _("Device is being updated"),
+            _("Do not unplug the device during the update."),
+            "emblem-synchronized"
+        );
+        progress_alert_view.get_style_context ().remove_class (Gtk.STYLE_CLASS_VIEW);
+
+        progress_view = new Gtk.Grid () {
+            margin = 24
+        };
+        progress_view.attach (progress_alert_view, 0, 0);
+
+        var no_devices_alert_view = new Granite.Widgets.AlertView (
+            _("No devices available"),
+            _("We're unable to provide updates for device firmware."),
+            "application-x-firmware"
+        );
```

Comment:
copy tweak

```suggestion
        var no_devices_alert_view = new Granite.Widgets.AlertView (
            _("No devices available"),
            _("Firmware updates are not supported on this or any connected devices."),
            "application-x-firmware"
        );
```

Reply:
@elementary/ux is there a better phrase here than "updatable"? It's not that there aren't any devices, or that there's no firmware, it's that there aren't and devices connected that are able to be updated.

Reply:
Thanks for your feedback. I'm always impressed how understandable you express things with few words 👀

Reply:
Maybe "No devices able to update"?